### PR TITLE
Add ch.bailu.gtk_meteo

### PR DESCRIPTION
**Pull request by bailuk**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
GTK Meteo: Weather forecast application based on [mapsforge-gtk](https://github.com/bailuk/mapsforge-gtk) and [java-gtk](https://github.com/bailuk/java-gtk). Forecast data is from [Meteorologisk institutt Norge](https://www.met.no/).

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. 

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: ch.bailu.gtk_meteo.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      ]
+    },
+    {
+      "name": "gtk-meteo",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/bailuk/gtk-meteo.git",
+          "branch": "543eb0a5ff40b57d70277fb6c7ac9b3e6f9fb431"
```
Comment (hfiguiere) [inline ch.bailu.gtk_meteo.json:25]: NEVER branch. This is a commit btw.

File: ch.bailu.gtk_meteo.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        "source /usr/lib/sdk/openjdk17/enable.sh && ./gradlew build",
+        "install -Dm755 ci/flatpak-run-app.sh /app/bin/gtk-meteo.sh",
+        "install -Dm644 build/libs/gtk-meteo-all.jar /app/lib/gtk-meteo.jar",
+        "install -Dm644 flatpak/ch.bailu.gtk_meteo.desktop /app/share/applications/ch.bailu.gtk_meteo.desktop",
+        "install -Dm644 flatpak/ch.bailu.gtk_meteo.metainfo.xml /app/share/metainfo/ch.bailu.gtk_meteo.metainfo.xml",
+        "install -Dm644 src/main/resources/svg/app-icon.svg /app/share/icons/hicolor/scalable/apps/ch.bailu.gtk_meteo.svg"
+      ]
+    }
+  ],
+  "finish-args": [
```
Comment (hfiguiere) [inline ch.bailu.gtk_meteo.json:40]: this should be above modules.

File: ch.bailu.gtk_meteo.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "--share=ipc",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--env=GIO_EXTRA_MODULES=/app/lib/gio/modules",
+    "--env=JAVA_HOME=/app/jre",
+    "--env=LIBO_FLATPAK=1",
+    "--env=PATH=/app/bin:/app/jre/bin",
+    "--filesystem=~/.config/gtk-meteo",
+    "--filesystem=~/.java/.userPrefs/gtk-meteo"
```
Comment (hfiguiere) [inline ch.bailu.gtk_meteo.json:51]: these two should be "persist" instead. 

also are you sure it's `.config` and not `$XDG_CONFIG_HOME` ?

File: ch.bailu.gtk_meteo.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    {
+      "name": "gtk-meteo",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/bailuk/gtk-meteo.git",
+          "branch": "543eb0a5ff40b57d70277fb6c7ac9b3e6f9fb431"
+        },
+        "gradle-sources.json"
```
Comment (hfiguiere) [inline ch.bailu.gtk_meteo.json:27]: this file is missing.

File: ch.bailu.gtk_meteo.json (hunk 1 above)
Comment (bailuk) [reply ch.bailu.gtk_meteo.json]: Changed to commit and updated upstream (see below)

File: ch.bailu.gtk_meteo.json (hunk 3 above)
Comment (bailuk) [reply ch.bailu.gtk_meteo.json]: Upstream now uses $XDG_CONFIG_HOME

**Comment by hfiguiere** [comment]
bot, build ch.bailu.gtk_meteo

File: ch.bailu.gtk_meteo.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--env=GIO_EXTRA_MODULES=/app/lib/gio/modules",
+    "--env=JAVA_HOME=/app/jre",
+    "--env=LIBO_FLATPAK=1",
+    "--env=PATH=/app/bin:/app/jre/bin"
```
Comment (bbhtt) [inline ch.bailu.gtk_meteo.json:19]:
```suggestion
    "--env=PATH=/app/bin:/usr/bin:/app/jre/bin"
```

Comment (bbhtt) [reply ch.bailu.gtk_meteo.json:19]: You are removing everything in runtime from PATH, this going to break massively.

**Comment by bailuk** [comment]
bot, build ch.bailu.gtk_meteo

**Comment by bbhtt** [comment]
The boxes above seems weirdly empty ?

image

**Comment by bailuk** [comment]
> The boxes above seems weirdly empty ?
> 
This is expected: Because there is no place selected yet. 
If the arrow-up button (top-right) on the map gets selected the boxes will be filled. And after that on next app start they should  be filled too.

**Pull request closed without merging**